analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs

```diff
@@ -335,6 +335,22 @@ public void InfiniteForLoop(string arg)
             }
         }
 
+        public void LoopWithIfsInside(bool condition)
+        {
+            var done = false;
+            var i = 0;
+            while (!done)
+            {
+                if (i <= 5)                 // Noncompliant FP
```

Comment:
Just to verify my understanding:
this will reach learning that i: [1,5] after the max loop so it will learn that it can never enter the else.
Thus this is an FP due to not visiting the loop more.
Is this correct?

Reply:
Yes. This is a bit unfortunate. But I don't see a good way to avoid these kinds of FPs without iterating loops more than twice.

Reply:
Was this already problem before (likely yes), or was it created with this PR?

Reply:
This was an issue before this PR already.